crates/kubelet/src/volume/persistentvolumeclaim.rs

```diff
@@ -0,0 +1,319 @@
+//
+// A volume with volumeMode: Filesystem is mounted into Pods into a directory.
+// If the volume is backed by a block device and the device is empty,
+// Kuberneretes creates a filesystem on the device before mounting it for the
+// first time.
+//
+// You can set the value of volumeMode to Block to use a volume as a raw block
+// device. Such volume is presented into a Pod as a block device, without any
+// filesystem on it. This mode is useful to provide a Pod the fastest possible
+// way to access a volume, without any filesystem layer between the Pod and the
+// volume. On the other hand, the application running in the Pod must know how
+// to handle a raw block device. See Raw Block Volume Support for an example on
+// how to use a volume with volumeMode: Block in a Pod.
+//
+// https://kubernetes.io/docs/concepts/storage/persistent-volumes/#volume-mode
+#[derive(Debug)]
+enum VolumeMode {
+    Block,
+    Filesystem,
+}
+
+impl FromStr for VolumeMode {
+    type Err = BadVolumeMode;
+
+    // defines what type of volume is required by the claim. The "filesystem"
+    // mode is implied when not included in the claim spec.
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Block" => Ok(VolumeMode::Block),
+            "Filesystem" => Ok(VolumeMode::Filesystem),
+            "" => Ok(VolumeMode::Filesystem),
+            _ => Err(BadVolumeMode),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct BadReclaimPolicy;
+
+impl fmt::Display for BadReclaimPolicy {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "bad reclaim policy")
+    }
+}
+
+impl Error for BadReclaimPolicy {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        None
+    }
+}
+
+#[derive(Debug)]
+enum ReclaimPolicy {
+    Delete,
+    Recycle,
+    Retain,
+}
+
+impl FromStr for ReclaimPolicy {
+    type Err = BadReclaimPolicy;
+
+    // defines what type of reclaim policy is requested. The "delete" mode is
+    // implied when not included in the storage class or persistent volume.
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Delete" => Ok(ReclaimPolicy::Delete),
+            "Recycle" => Ok(ReclaimPolicy::Recycle),
+            "Retain" => Ok(ReclaimPolicy::Retain),
+            "" => Ok(ReclaimPolicy::Delete),
+            _ => Err(BadReclaimPolicy),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct BadAccessMode;
+
+impl fmt::Display for BadAccessMode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "bad access mode")
+    }
+}
+
+impl Error for BadAccessMode {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        None
+    }
+}
+
+#[derive(Debug)]
+enum AccessMode {
+    ReadOnlyMany,
+    ReadWriteMany,
+    ReadWriteOnce,
+}
+
+impl FromStr for AccessMode {
+    type Err = BadAccessMode;
+
+    // defines what type of reclaim policy is requested. The "delete" mode is
```

Comment:
This looks like copypasta

Reply:
Is there something I should address here?

Reply:
For context, yes this is copypasta from the rust docs. Reference implementation: https://doc.rust-lang.org/std/str/trait.FromStr.html

Reply:
I think the issue might be `s/reclaim policy/access mode`?

Reply:
Yep, it seemed like it was from another struct.

Reply:
:headsmack: I see that now. Will address tomorrow :)